### DISCUSSION TITLE
How to Turn Off Quick Suggestions

### DISCUSSION THREAD

**Discussion by jasc2v8** [mark-wiemer/ahkpp#68] [General]
When you type your AHK code, the Intellisense will auto popup quick suggestions by default.
That is fine, but I get dozens of suggestions from other languages.
I've tried a lot of things to turn off all other languages except AHK but haven't figured it out yet.
Therefore, I turn off ALL quick suggestions.
I can still type a few letters then press Ctrl+Space to get quick suggestions.
This makes VSCode much easier to use for my personal preference.

How to Turn Off Quick Suggestions in VSCode:
1. Press F1, type "open user settings", search for "edit.quick"
2. Next to Editor.QuickSuggestions, press the link to "Edit in settinsg.json"
3. Search for and change from "null" to "false":  "editor.quickSuggestions": false,
4. Press F1, type "reload", choose "Developer: Reload Window"

Read the VSCode help pages if you wish to change in your workspace versus user settins.
Enjoy!

**Comment by jasc2v8** [comment]
I forgot an important one!
The default is to show a quick suggestion when the spacebar it typed.
This disables that action.
You can always press Ctrl+Spcce to get quick suggestions.
```
editor.suggestOnTriggerCharacters": false,
```

**Reply by fade2gray** [reply]
Excellent - I thought it was a 'feature' of Intellisense.

@mark-wiemer I thought toggle IntelliSense on/off had been fixed or hasn't it been implemented yet?

**Reply by mark-wiemer** [reply]
To be honest, I'm not 100% sure what the IS toggle is supposed to do. I do know that it changes suggestions a bit, but I've never looked into it, sorry

**Reply by fade2gray** [reply]
I was referring to #48 - are we talking at cross purposes?

**Comment by mark-wiemer** [comment]
Cool, I know everyone has personal preferences when it comes to quick suggestions. I'm glad you found something that works for you, let me know if you have recommendations for better trigger characters that would be less annoying

**Reply by jasc2v8** [reply]
I think the trigger chars for AHK are perfect!
F1, "Preferences: Open Keyboard Shortcuts", search for the word "Trigger"
I have: Tab, Space, Ctrl+Space, Ctrl+Shift+Space, and Ctrl+I (the letter "i")

I'm really enjoying VScode with these settings for AHK:
```
"editor.quickSuggestions": false,                    //don't auto show quick suggestions
"editor.suggestOnTriggerCharacters": false,          //don't trigger on spacebar
```
I want to again complement you on a great extension - thank you!
😄
